### ISSUE TITLE
Support object methods as factories

### ISSUE THREAD

**Issue by mnapoli** [PHP-DI/PHP-DI#285]
```php
return [
    'Foo' => [new FooFactory(), 'create'],
    // these examples are not supported:
    'Foo' => [DI\get('FooFactory'), 'create'],
    'Foo' => ['FooFactory', 'create'],
];
```

Those examples are actually in the documentation, but they do not work. They need to be supported.

For the record they are supported in `$container->call()`.

**Comment by mnapoli** [comment]
Would be covered by #197

**Comment by mnapoli** [comment]
Was closed by #308
